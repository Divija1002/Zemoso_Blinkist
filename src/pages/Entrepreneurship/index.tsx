import axios from "axios";
import React, { useEffect, useState, Fragment } from "react";
import { Box, Typography, TextField } from "@mui/material";
import SearchIcon from "@mui/icons-material/Search";
import Banner from "../../components/molecules/Banner";
import BCard from "../../components/molecules/BCard";
import BasicTemplate from "../../components/templates/BasicTemplate";
import {
  FEATURED_AUDIO_BLINK,
  JUST_ADDED,
  TRENDING_BOOKS,
} from "../../components/Constants";

interface BookType {
  id: number;
  image: string;
  title: string;
  author: string[];
  readTime: number;
  readersCount?: string;
  status: string;
}

const Entrepreneurship = () => {
  const TrendingBooks = TRENDING_BOOKS;

  const JustAdded = JUST_ADDED;

  const FeaturedAudioBlink = FEATURED_AUDIO_BLINK;

  const [books, setBooks] = useState<BookType[]>([]);

  const getBooks = () => {
    axios
      .get("http://localhost:8000/bookInfo")
      .then((response) => {
        console.log(response);
        setBooks(response.data);
      })
      .catch((error) => {
        console.log(error);
      });
  };

  useEffect(() => {
    getBooks();
    const interval = setInterval(() => {
      getBooks();
    }, 1000);
    return () => clearInterval(interval);
  }, []);

  const [textInput, setTextInput] = useState("");

  const handleTextInputChange = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    setTextInput(event.target.value);
  };

  return (
    <BasicTemplate
      content={
        <Fragment>
          <Banner />
          <Box
            sx={{
              paddingLeft: "266px",
              paddingTop: "58px",
              display: "flex",
              gap: "25px",
              alignItems: "flex-end",
            }}
          >
            <SearchIcon />
            <TextField
              id="standard-basic"
              label="Search by title or author"
              variant="standard"
              style={{ width: "658px" }}
              value={textInput}
              onChange={handleTextInputChange}
            />
          </Box>
          {textInput === "" ? (
            <Fragment>
              <Box
                sx={{
                  paddingLeft: "264px",
                  paddingTop: "80px",
                }}
              >
                <Typography variant="h3" sx={{ color: "#03314B" }}>
                  Trending blinks
                </Typography>
              </Box>
              <Box
                sx={{
                  paddingX: "259px",
                  paddingTop: "24px",
                  display: "flex",
                  flexWrap: "wrap",
                }}
              >
                {books.map((book) =>
                  TrendingBooks.includes(book.title) ? (
                    <BCard book={book} trending={true}></BCard>
                  ) : null
                )}
              </Box>
              <Box
                sx={{
                  paddingLeft: "264px",
                  paddingTop: "80px",
                }}
              >
                <Typography variant="h3" sx={{ color: "#03314B" }}>
                  Just added
                </Typography>
              </Box>
              <Box
                sx={{
                  paddingX: "259px",
                  paddingTop: "25px",
                  display: "flex",
                  flexWrap: "wrap",
                }}
              >
                {books.map((book) =>
                  JustAdded.includes(book.title) ? (
                    <BCard book={book} justAdded={true}></BCard>
                  ) : null
                )}
              </Box>
              <Box
                sx={{
                  paddingLeft: "264px",
                  paddingTop: "80px",
                }}
              >
                <Typography variant="h3" sx={{ color: "#03314B" }}>
                  Featured audio blinks
                </Typography>
              </Box>
              <Box
                sx={{
                  paddingX: "259px",
                  paddingTop: "25px",
                  display: "flex",
                  flexWrap: "wrap",
                }}
              >
                {books.map((book) =>
                  FeaturedAudioBlink.includes(book.title) ? (
                    <BCard book={book} featured={true}></BCard>
                  ) : null
                )}
              </Box>
            </Fragment>
          ) : (
            <Box
              sx={{
                paddingX: "259px",
                paddingTop: "80px",
                display: "flex",
                flexWrap: "wrap",
              }}
            >
              {books.map((book) =>
                book.title.toLowerCase().indexOf(textInput) > -1 ||
                book.author.indexOf(textInput) > -1 ? (
                  <BCard book={book}></BCard>
                ) : null
              )}
            </Box>
          )}
        </Fragment>
      }
    />
  );
};

export default Entrepreneurship;
