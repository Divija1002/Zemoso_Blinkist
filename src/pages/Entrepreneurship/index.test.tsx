import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Entrepreneurship from ".";

describe("Entrepreneurship snapshot", () => {
  it("Entrpreneurship page testing", () => {
    const renderedComponent = renderer.create(
      <BrowserRouter>
        <Entrepreneurship />
      </BrowserRouter>
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
