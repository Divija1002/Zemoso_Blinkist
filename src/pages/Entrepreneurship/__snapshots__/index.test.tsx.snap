// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entrepreneurship snapshot Entrpreneurship page testing 1`] = `
Array [
  <div
    className="MuiBox-root css-1mvk6tf"
  >
    <img
      alt="logo"
      className="MuiBox-root css-1parqao"
      onClick={[Function]}
      src="images/Blinklist 1.png"
    />
    <img
      alt="logo"
      className="MuiBox-root css-1oqfcha"
      src="images/Vector.png"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-sqc3pa-MuiTypography-root"
    >
      Explore
    </p>
    <img
      alt="arrow"
      className="MuiBox-root css-1s8et9c"
      onClick={[Function]}
      src="images/Vector1.png"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-kfnt82-MuiTypography-root"
    >
      My Library
    </p>
    <div
      className="MuiBox-root css-xo4s73"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-4yn7uo-MuiTypography-root"
      />
    </div>
    <img
      alt="arrow"
      className="MuiBox-root css-1s8et9c"
      onClick={[Function]}
      src="images/Vector1.png"
    />
  </div>,
  <div
    className="MuiBox-root css-1kqchqf"
  />,
  <div
    className="MuiBox-root css-l0eg0a"
  >
    <div
      className="MuiBox-root css-k008qs"
    >
      <div
        className="MuiBox-root css-j7qwjs"
      >
        <div
          className="MuiBox-root css-10ybs0l"
        >
          <h1
            className="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
          >
            Explore Books on entrepreneurship
          </h1>
        </div>
        <div
          className="MuiBox-root css-1t1xgc0"
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Everything you need to know about thriving on a shoestring budget, making your first million, and hiring right from the start.
          </h6>
        </div>
      </div>
      <img
        alt="banner"
        className="MuiBox-root css-isrw60"
        src="images/banner.png"
      />
    </div>
  </div>,
  <div
    className="MuiBox-root css-wd6d1g"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SearchIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      style={
        Object {
          "width": "658px",
        }
      }
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="standard-basic"
        id="standard-basic-label"
      >
        Search by title or author
      </label>
      <div
        className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={false}
          id="standard-basic"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="MuiBox-root css-1k9euxb"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3 css-1jybd6-MuiTypography-root"
    >
      Trending blinks
    </h3>
  </div>,
  <div
    className="MuiBox-root css-pob5tk"
  />,
  <div
    className="MuiBox-root css-1k9euxb"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3 css-1jybd6-MuiTypography-root"
    >
      Just added
    </h3>
  </div>,
  <div
    className="MuiBox-root css-7n55bm"
  />,
  <div
    className="MuiBox-root css-1k9euxb"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3 css-1jybd6-MuiTypography-root"
    >
      Featured audio blinks
    </h3>
  </div>,
  <div
    className="MuiBox-root css-7n55bm"
  />,
  <div
    className="MuiBox-root css-12ntjjl"
  >
    <div
      className="MuiBox-root css-1m9j77u"
    >
      <div
        className="MuiBox-root css-1ivgbss"
      >
        <img
          alt="logo"
          className="MuiBox-root css-15p9n5u"
          src="images/Blinklist 1.png"
        />
        <div
          className="MuiBox-root css-gnbw2f"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-147xq6v-MuiTypography-root"
          >
            Big ideas in small packages Start learnign now
          </p>
        </div>
      </div>
      <div
        className="MuiBox-root css-i09f21"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1ly8ndt-MuiTypography-root"
        >
          Editorial
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          Book lists
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          What is Nonfiction?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          What to read next?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          Benefits of reading
        </p>
      </div>
      <div
        className="MuiBox-root css-i09f21"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1ly8ndt-MuiTypography-root"
        >
          Useful links
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          Pricing
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          Blinkist business
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          Gift cards
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          Blinkist magaine
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          Contact & help
        </p>
      </div>
      <div
        className="MuiBox-root css-i09f21"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1ly8ndt-MuiTypography-root"
        >
          Company
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          About
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          Careers
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          partners
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1a3wegw-MuiTypography-root"
        >
          Code of Conduct
        </p>
      </div>
    </div>
    <div
      className="MuiBox-root css-1io1hfo"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-89tq0x-MuiTypography-root"
      >
        Â© Blinkist 2021 Sitemap | Imprint | Terms of Service | Privacy Policies
      </p>
    </div>
  </div>,
]
`;
